x-pack/plugins/cloud_security_posture/public/pages/benchmarks/benchmarks_table.tsx

```diff
@@ -5,60 +5,138 @@
+  return (
+    <EuiLink
+      href={http.basePath.prepend(pagePathGetters.policy_details({ policyId }).join(''))}
+      title={name}
+    >
+      {name}
+    </EuiLink>
+  );
+};
+
+const BENCHMARKS_TABLE_COLUMNS: Array<EuiBasicTableColumn<Benchmark>> = [
   {
-    field: 'integration_name',
-    name: TABLE_COLUMN_HEADERS.INTEGRATION_NAME,
-    dataType: 'string',
+    field: 'package_policy.name',
+    name: TABLE_COLUMN_HEADERS.BENCHMARK_RULES,
+    render: (v, r) => (
+      <Link to={`/benchmark/${r.package_policy.id}/${r.package_policy.policy_id}/rules`} title={v}>
+        {v}
+      </Link>
+    ),
+    truncateText: true,
   },
   {
-    field: 'benchmark',
-    name: TABLE_COLUMN_HEADERS.BENCHMARK,
-    dataType: 'string',
+    field: 'package_policy.rules', // TODO: add fields
+    name: TABLE_COLUMN_HEADERS.ACTIVE_RULES,
+    truncateText: true,
+    // render: (benchmarkIntegration) =>
+    //   `${benchmarkIntegration.rules.active} of ${benchmarkIntegration.rules.total}`,
   },
   {
-    render: (benchmarkIntegration: CspBenchmarkIntegration) =>
-      `${benchmarkIntegration.rules.active} of ${benchmarkIntegration.rules.total}`,
-    name: TABLE_COLUMN_HEADERS.ACTIVE_RULES,
+    field: 'package_policy.package.title',
+    name: TABLE_COLUMN_HEADERS.BENCHMARK,
+    dataType: 'string',
+    truncateText: true,
   },
+
   {
     field: 'agent_policy.name',
     name: TABLE_COLUMN_HEADERS.AGENT_POLICY,
-    dataType: 'string',
+    render: (v, r) => <AgentPolicyButtonLink name={v} id={r.agent_policy.id} />,
+    truncateText: true,
   },
   {
-    field: 'agent_policy.number_of_agents',
+    field: 'agent_policy.agents',
     name: TABLE_COLUMN_HEADERS.NUMBER_OF_AGENTS,
-    dataType: 'number',
+    truncateText: true,
   },
   {
-    field: 'created_by',
+    field: 'package_policy.created_by',
     name: TABLE_COLUMN_HEADERS.CREATED_BY,
     dataType: 'string',
+    truncateText: true,
   },
+
   {
-    field: 'created_at',
+    field: 'package_policy.created_at',
     name: TABLE_COLUMN_HEADERS.CREATED_AT,
     dataType: 'date',
-    render: (date: CspBenchmarkIntegration['created_at']) => moment(date).fromNow(),
+    truncateText: true,
+    render: (date: Benchmark['package_policy']['created_at']) => moment(date).fromNow(),
   },
 ];
 
-export const BenchmarksTable = ({ benchmarks, ...rest }: BenchmarksTableProps) => (
-  <EuiBasicTable
-    data-test-subj={rest['data-test-subj']}
-    items={benchmarks}
-    columns={BENCHMARKS_TABLE_COLUMNS}
-  />
-);
+export const BenchmarksTable = ({
+  benchmarks,
+  pageIndex,
+  pageSize,
+  totalItemCount,
+  loading,
+  error,
+  setQuery,
+  noItemsMessage,
+  ...rest
+}: BenchmarksTableProps) => {
+  const history = useHistory();
+
+  const getRowProps: EuiBasicTableProps<Benchmark>['rowProps'] = (item) => ({
+    onClick: (evt: MouseEvent) =>
+      // Navigate to rules page unless we already clicked a link
+      !(evt.target instanceof HTMLAnchorElement) &&
```

Comment:
this is indeed hacky 😆 

explicitly handling `stopPropogation` is definitely better, i'll change this.

initially avoided it because in this case we do want it to propagate as it's a link to an external app, which is handled by `RedirectAppLinks`. so we'd need to do that as well, but yeah, worth it.